app/client/ui/AdminPanel.ts

```diff
@@ -441,28 +415,33 @@ in the future as session IDs generated since v1.1.16 are inherently cryptographi
         case State.AVAILABLE: return t('Newer version available');
         case State.ERROR: return '❌ ' + t('Error checking for updates');
         case State.STALE: {
-          const lastCheck = use(lastCheckDate);
-          return t('Last checked {{time}}', {time: lastCheck ? getTimeFromNow(lastCheck) : 'n/a'});
+          const lastCheck = latestVersionAvailable.get()?.dateChecked;
+          return lastCheck ?
+            t('Last checked {{time}}', {time: getTimeFromNow(lastCheck)})
+            : t('No record of last version check');
         }
       }
     });
 
-    // Now trigger the initial state, by checking if we should auto-check.
-    if (onLoad.get()) {
-      actions.checkForUpdates().catch(reportError);
-    } else {
-      if (comparison.get() === 'new') {
-        state.set(State.AVAILABLE);
-      } else if (comparison.get() === 'old') {
+    // Now trigger the initial state
+    const lastCheck = latestVersionAvailable.get()?.dateChecked;
+    if (lastCheck) {
+      if (Date.now() - lastCheck > 14*24*60*60*1000) {
```

Comment:
Move constant to declared constant near start of file. Mention units, give docstring.